apps/omg_watcher_rpc/lib/web/validators/constraints.ex

```diff
@@ -12,30 +12,30 @@
 # See the License for the specific language governing permissions and
 # limitations under the License.
 
-defmodule OMG.WatcherRPC.Web.Validator.Constrains do
+defmodule OMG.WatcherRPC.Web.Validator.Constraints do
   @moduledoc """
   Validates `/transaction.all` query parameters
   """
 
   import OMG.Utils.HttpRPC.Validator.Base
 
   @doc """
-  Validates possible query constrains, stops on first error.
+  Validates possible query constraints, stops on first error.
   """
   @spec parse(%{binary() => any()}) :: {:ok, Keyword.t()} | {:error, any()}
   def parse(params) do
-    constrains = [
+    constraints = [
       address: [:address, :optional],
       blknum: [:pos_integer, :optional],
       metadata: [:hash, :optional],
       limit: [:pos_integer, :optional],
       page: [:pos_integer, :optional]
     ]
 
-    constrains
-    |> Enum.reduce_while({:ok, []}, fn {constrain, validators}, {:ok, list} ->
-      with {:ok, value} when not is_nil(value) <- expect(params, Atom.to_string(constrain), validators) do
-        {:cont, {:ok, [{constrain, value} | list]}}
+    constraints
+    |> Enum.reduce_while({:ok, []}, fn {constraint, validators}, {:ok, list} ->
```

Comment:
`Enum.reduce_while(constraints, {:ok, []}, fn {constraint, validators}, {:ok, list} ->`

Reply:
also, this can be written much better with a case clause
```
Enum.reduce_while(constraints, {:ok, []}, fn {constraint, validators}, {:ok, list} ->
  case expect(params, Atom.to_string(constraint), validators) do
    {:ok, nil} -> {:cont, {:ok, list}}
    {:ok, value} -> {:cont, {:ok, [{constraint, value} | list]}}
    error -> {:halt, error}
  end
end
```

Reply:
Indeed case is more readable - :+1: